packages/bloc/lib/src/bloc.dart

```diff
@@ -7,6 +7,16 @@ part 'bloc_base.dart';
 part 'bloc_overrides.dart';
 part 'emitter.dart';
 
+/// An [ErrorSink] that supports adding events.
+///
+/// Multiple events can be reported to the sink via `add`.
+abstract class EventSink<Event extends Object?> implements ErrorSink {
```

Comment:
[EventSink is an existing interface](https://api.dart.dev/stable/2.14.4/dart-async/EventSink-class.html) from the `dart:async` library

Reply:
Yup I went back and forth on it but ultimately decided it made more sense to have a custom EventSink so that it implements ErrorSink rather than two unrelated classes that happen to have the same API. Also if we go back to implementing EventSink from dart:async we’ll be prone to the false positives with the close_sinks lint rule.

Reply:
> it made more sense to have a custom EventSink
I totally agree! One thing is I am a bit worried about is the name clash :) Currently was testing this branch and had to revert to named imports in order to use both EventSinks.

Reply:
> I totally agree! One thing is I am a bit worried about is the name clash :) Currently was testing this branch and had to revert to named imports in order to use both EventSinks.

Yes, if you want to use both you’d need named imports but is there a common use case in which you’d need to import both interfaces within the same file?

> Also, having two classes with the same name and relatively the same meaning from two different libraries can be confusing.

Agreed, but I think it’d be more confusing to have Bloc implement both EventSink and ErrorSink where the two come from different libraries but have the same API.

Reply:
@purplenoodlesoop I suppose we can rename Sink to something else like Pipe, Vent, or Chute 🤷‍♂️

What do you think?

Reply:
> Agreed, but I think it’d be more confusing to have Bloc implement both EventSink and ErrorSink where the two come from different libraries but have the same API.

100%, I agree that a custom interface is needed.

> I suppose we can rename Sink to something else like Channel, Consumer, Receiver, or Pipe 

What about `BlocEventSink`?

Reply:
Yeah that works as well. My only concern is from reading the class name I’d assume it implements EventSink from dart:async 😅

Reply:
Whoops, indeed. Namings are hard 🥲

Reply:
Yeah I know I’m terrible at naming 😅

I will update to BlocEventSink for now though since I agree that’s a better solution.

Reply:
> Yeah I know I’m terrible at naming 😅

No no, namings are indeed tricky :) Or we all are terrible at them lol